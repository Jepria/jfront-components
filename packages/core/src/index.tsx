export * from '@jfront-ui/app'
export * from '@jfront-ui/datepicker'
export * from '@jfront-ui/form'
//export * from '@jfront-ui/grid'
export * from '@jfront-ui/hooks'
//export * from '@jfront-ui/input'
export * from '@jfront-ui/layout'
//export * from '@jfront-ui/pagingbar'
//export * from '@jfront-ui/tabpanel'
//export * from '@jfront-ui/toolbar'
