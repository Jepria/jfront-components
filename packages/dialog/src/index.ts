export * from './SimpleDialog'
export * from './AlertDialog'
export * from './FormDialog'
export * from './ErrorDialog'
export * from './WarningDialog'