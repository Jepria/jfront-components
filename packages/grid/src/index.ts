export * from './common'
export * from './jepgrid'
