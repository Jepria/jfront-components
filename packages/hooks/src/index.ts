export * from "./useOnClickOutside"
export * from "./useRouteMapping"
export * from "./useTree/"
export * from "./useTree/types"
export * from "./useFilter"
