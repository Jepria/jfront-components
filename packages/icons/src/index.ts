export * from "./Add"
export * from "./Delete"
export * from "./Edit"
export * from "./Exclamation"
export * from "./First"
export * from "./Last"
export * from "./Loading"
export * from "./Next"
export * from "./OpenIcon"
export * from "./Prev"
export * from "./Refresh"
export * from "./Save"
export * from "./Search"
export * from "./SettingsSmall"
export * from "./Split"
export * from "./View"
