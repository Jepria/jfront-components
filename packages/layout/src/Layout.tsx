import styled from "styled-components"

export const Column = styled.div`
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -ms-flex-direction: column;
  flex-direction: column;
`

export const Row = styled.div`
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -ms-flex-direction: row;
  flex-direction: row;
`
