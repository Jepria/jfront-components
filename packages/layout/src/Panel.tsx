import styled from "styled-components"
import React from "react"

const StyledPanel = styled.section`
  box-sizing: border-box;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -ms-flex-direction: column;
  flex-direction: column;
`

const StyledHeader = styled.header`
  box-sizing: border-box;
  height: auto;
  width: 100%;
`

const StyledContent = styled.section`
  box-sizing: border-box;
  width: 100%;
  overflow: auto;
  -webkit-box-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
`

const StyledFooter = styled.footer`
  box-sizing: border-box;
  height: auto;
  width: 100%;
`

export type Panel = React.FC<React.HTMLAttributes<HTMLDivElement>> & {
  Header: React.FC<React.HTMLAttributes<HTMLDivElement>>
  Content: React.FC<React.HTMLAttributes<HTMLDivElement>>
  Footer: React.FC<React.HTMLAttributes<HTMLDivElement>>
}

export const Panel: Panel = (props) => {
  return <StyledPanel {...props}>{props.children}</StyledPanel>
}

Panel.Header = (props) => {
  return <StyledHeader {...props}>{props.children}</StyledHeader>
}

Panel.Content = (props) => {
  return <StyledContent {...props}>{props.children}</StyledContent>
}

Panel.Footer = (props) => {
  return <StyledFooter {...props}>{props.children}</StyledFooter>
}
