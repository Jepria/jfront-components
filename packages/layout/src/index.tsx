import styled from 'styled-components';

export const FormContainer = styled.div`
  padding: 5px;
`;

export const VerticalLayout = styled.div`
  width: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
`;

export const HorizontalLayout = styled.div`
  width: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
      -ms-flex-direction: row;
          flex-direction: row;
`;

export const Page = styled.div`
  box-sizing: border-box;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
`;

export const Header = styled.header`
  box-sizing: border-box;
  height: auto;
  width: 100%;
`;

export const Content = styled.section`
  box-sizing: border-box;
  width: 100%;
  overflow: auto;
  -webkit-box-flex: 1;
      -ms-flex: 1;
          flex: 1;
  -ms-flex-positive: 1;
      flex-grow: 1;
`;

export const Footer = styled.footer`
  box-sizing: border-box;
  height: auto;
  width: 100%;
`;
