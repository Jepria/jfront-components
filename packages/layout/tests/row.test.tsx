import React from "react"
import { render } from "@testing-library/react"
import { Column, Panel, Row } from "../src"

test("Checking for the existence of an element Tab", () => {
  render(
    <>
      <Row style={{ height: "100%" }}>
        <Column style={{ flexGrow: 1, backgroundColor: "green" }}>
          <Panel />
        </Column>
        <Column style={{ flexGrow: 3 }}>
          <Panel>
            <Panel.Header
              style={{ backgroundColor: "yellow", minHeight: "30px" }}
            />
            <Panel.Content style={{ backgroundColor: "red" }} />
            <Panel.Footer
              style={{ backgroundColor: "blue", minHeight: "30px" }}
            />
          </Panel>
        </Column>
      </Row>
    </>,
  )
})

test("Matches snapshot ", () => {
  const { asFragment } = render(
    <>
      <Row style={{ height: "100%" }}>
        <Column style={{ flexGrow: 1, backgroundColor: "green" }}>
          <Panel />
        </Column>
        <Column style={{ flexGrow: 3 }}>
          <Panel>
            <Panel.Header
              style={{ backgroundColor: "yellow", minHeight: "30px" }}
            />
            <Panel.Content style={{ backgroundColor: "red" }} />
            <Panel.Footer
              style={{ backgroundColor: "blue", minHeight: "30px" }}
            />
          </Panel>
        </Column>
      </Row>
    </>,
  )
  expect(asFragment()).toMatchSnapshot()
})
