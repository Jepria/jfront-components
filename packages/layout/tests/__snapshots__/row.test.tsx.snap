// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot  1`] = `
<DocumentFragment>
  <div
    class="sc-AxirZ jsFaQM"
    style="height: 100%;"
  >
    <div
      class="sc-AxjAm fDqZvr"
      style="flex-grow: 1; background-color: green;"
    >
      <section
        class="sc-AxiKw gGQAYG"
      />
    </div>
    <div
      class="sc-AxjAm fDqZvr"
      style="flex-grow: 3;"
    >
      <section
        class="sc-AxiKw gGQAYG"
      >
        <header
          class="sc-AxhCb UORoV"
          style="background-color: yellow; min-height: 30px;"
        />
        <section
          class="sc-AxhUy ereXFW"
          style="background-color: red;"
        />
        <footer
          class="sc-AxgMl jwgXaO"
          style="background-color: blue; min-height: 30px;"
        />
      </section>
    </div>
  </div>
</DocumentFragment>
`;
